src/hooks/useAuthFlow.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useContext } from 'react'
+import { ProviderType } from '@dcl/schemas'
+import { getConfiguration, connection } from 'decentraland-connect'
+import { FeatureFlagsContext, FeatureFlagsKeys } from '../components/FeatureFlagsProvider'
+import { fetchProfile } from '../modules/profile'
+import { locations } from '../shared/locations'
+import { isProfileComplete } from '../shared/profile'
+import { useNavigateWithSearchParams } from './navigation'
+import { useAfterLoginRedirection } from './redirection'
+import { useTargetConfig } from './targetConfig'
+
+export const useAuthFlow = () => {
+  const navigate = useNavigateWithSearchParams()
+  const { url: redirectTo } = useAfterLoginRedirection()
+  const { flags } = useContext(FeatureFlagsContext)
```

Comment:
Be aware that the `connectToMagic` that you're abstracting should be executed only after the FFs are loaded. I would give the responsibility of checking this to the flow that is using the `useAuthFlow` hook. The same goes for the other flag.
I think that it might be a bit confusing to use the flags context here taking into consideration that all of these function must be executed only when the flags are loaded.